http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java

```diff
@@ -407,6 +409,25 @@ public HttpClientConfig setKerberosRemoteServiceName(String serviceName)
         return this;
     }
 
+    public enum NameType
+    {
+        HOSTBASED_SERVICE,
+        USER_NAME
+    }
+
+    public NameType getKerberosNameType()
+    {
+        return kerberosNameType;
+    }
+
+    @Config("http-client.authentication.krb5.name-type")
+    @ConfigDescription("Set kerberos GSS name type")
+    public HttpClientConfig setKerberosNameType(String nameType)
```

Comment:
Make the config method take the enum directly:
```java
public HttpClientConfig setKerberosNameType(KerberosNameType nameType)
```
The config system will do the conversion and give a good error message on failure.